src/input/mod.rs

```diff
@@ -1,18 +1,70 @@
+#[cfg(feature = "async")]
+use crate::WaitableInputSwitch;
 use crate::{ActiveHigh, ActiveLow, InputSwitch, Switch};
-use embedded_hal::digital::v2::InputPin;
+use embedded_hal::digital::{ErrorType, InputPin};
+#[cfg(feature = "async")]
+use embedded_hal_async::digital::Wait;
 
 impl<T: InputPin> InputSwitch for Switch<T, ActiveHigh> {
-    type Error = <T as InputPin>::Error;
+    type Error = <T as ErrorType>::Error;
 
     fn is_active(&self) -> Result<bool, Self::Error> {
-        self.pin.is_high()
+        self.pin.borrow_mut().is_high()
```

Comment:
This can panic without the user expecting it. I told you earlier that internal mutability is complicated...

RefCell::borrow_mut - Panics if the value is currently borrowed. For a non-panicking variant, use [try_borrow_mut](https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.try_borrow_mut).